common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java

```diff
@@ -47,13 +47,20 @@ public static int roundNumberOfBytesToNearestWord(int numBytes) {
   public static boolean arrayEquals(
       Object leftBase, long leftOffset, Object rightBase, long rightOffset, final long length) {
     int i = 0;
-    while (i <= length - 8) {
-      if (Platform.getLong(leftBase, leftOffset + i) !=
-        Platform.getLong(rightBase, rightOffset + i)) {
-        return false;
-      }
-      i += 8;
-    }
+
+      // This attempts to speed up the memcmp type of operation, but there is no way
+      // to guarantee that the offsets will be on a word boundary in order to use
+      // Platform.getLong
```

Comment:
It should happen 1/8 of the time for left, and 1/8 of the time for right, and those times have to match up at the same time, so 1/64, and I don't believe it actually happens that often.  We can leave the current behavior for non-SPARC if you want.   

Just FYI though, even though it "works" on non-SPARC, it is still slower than aligned accesses as manipulations are going on under the hood to make it "work".